### ISSUE TITLE
Charging above sailing limit when offline

### ISSUE THREAD

**Issue by Bachsau** [davidwernhart/AlDente-Charge-Limiter#697]
**Describe the bug**
I just found out that if I enable sailing mode, then shutdown or restart the system, it starts charging again. This is an Intel based system, and the SMC limiting does work. However it would be more reasonable if AlDente would set this limit to the lower boundary when in sailing mode. This way, the MacBook Pro would not start charging again, as long as the battery is in the sailing range. If AlDente then detects that it has fallen below that limit, as soon as it's running again, it should change the limit to the upper boundary and top up the battery. As soon as that state is reached, it should again write the lower limit to the SMC.

**To Reproduce**
1. Set a maximum of 70%
2. Set a 10% sailing range
3. Let the battery drain to 65%
4. Shutdown the system while the charger is connected
5. Notice that the battery has been re-charged to 70%

**Expected behavior**
The battery not being re-charged as long as it doesn't fall below 60%, even when the system is off.

**MacBook (please complete the following information):**
 - MacBook Pro Early 2015
 - macOS 12.4
 - AlDente Pro
 - AlDente Version 1.17.1
 - List of all enabled settings in AlDente: Check for updates, start on login, sailing mode, disable sleep-mode, hardware-battery percentage
 - List of all disabled settings in AlDente: Heat protection

**Additional context**
I know this would also result in the battery only being charged up to 60% when the system is off, but since a turned-off system will not really drain the battery, that won't be a problem, and if it's shut down while already down below the lower limit, it would charge to the upper limit, because AlDente would have updated the SMC as soon as it noticed the battery had fallen below the lower limit.

In short: Don't just set the SMC to the upper limit defined by the user, but update it every time a sailing limit is reached. If the battery has reached the upper limit, set it to the lower boundary, and as soon as that value is reached, move it up again.

**Comment by MatthiasKerbl** [comment]
Hi @Bachsau,

Thank you for this thorough description and the good idea for an improvement.
We have just released AlDente 1.22 which includes a fix for this issue. 

Best,
 Matthias
